docs/source/deployment/lambda.mdx

```diff
@@ -284,42 +252,177 @@ If you ever want to remove the S3 bucket or Lambda functions that Serverless cre
  serverless remove
 ```
 
-## Customizing HTTP behavior
+## Middleware
 
-The `@as-integrations/aws-lambda` package is compatible with the following event types `APIGatewayProxyEvent`, `APIGatewayProxyEventV2`, `ALBEvent`. This supports a wide range of services like API Gateway HTTP Proxy APIs, API Gateway REST Proxy APIs, Lambda Function URLs, and Application Load Balancers. However, it does not let you customize HTTP behavior directly or support other AWS products that invoke Lambda functions (e.g., S3 or DynamoDB).
+In order to implement event and result mutations, type-safe middleware can be passed to the `startServerAndCreateLambdaHandler` call. The API is as follows:
 
-If you want to customize your HTTP behavior, you can couple Apollo Server's Express integration (i.e., [`expressMiddleware`](../api/express-middleware)) with the [`@vendia/serverless-express`](https://github.com/vendia/serverless-express) package. The `@vendia/serverless-express` library translates between Lambda events and Express requests. Despite their similar names, the Serverless CLI and the `@vendia/serverless-express` package are unrelated.
+```ts
 
-You can update your Apollo Server setup to the following to have a fully functioning Lambda server that works in a variety of AWS features:
+import { middleware, startServerAndCreateLambdaHandler, handlers } from "@as-integrations/aws-lambda";
 
-<MultiCodeBlock>
+const requestHandler = handlers.createAPIGatewayProxyEventV2RequestHandler()
+
+// All middleware must be async and will be strictly typed based on the `requestHandler`'s types
+const middlewareFn: middleware.MiddlewareFn<typeof requestHandler> = async (event) => {
+  // event is intended to be mutable and can be updated here
+
+  // optional result mutation, otherwise function returns `Promise<void>`
+  return async (result) => {
+    // result is intended to be mutable and can be updated here
+
+    // `Promise<void>` return
+  }
+}
+
+startServerAndCreateLambdaHandler(..., ..., {
```

Comment:
Eventually we'll be able to prettify code blocks in our docs (https://github.com/apollographql/apollo-server/pull/6914) so I'd like to have valid syntax wherever possible.
```suggestion
startServerAndCreateLambdaHandler(server, requestHandler, {
```